scarb/tests/e2e/build_starknet_contract.rs

```diff
@@ -6,39 +6,66 @@ use assert_fs::prelude::*;
 use cairo_lang_starknet::casm_contract_class::CasmContractClass;
 use cairo_lang_starknet::contract_class::ContractClass;
 use indoc::{formatdoc, indoc};
+use once_cell::sync::Lazy;
 use predicates::prelude::*;
 use serde::de::DeserializeOwned;
 
 use crate::support::command::Scarb;
 use crate::support::fsx::ChildPathEx;
 use crate::support::project_builder::ProjectBuilder;
 
-const BALANCE_CONTRACT: &str = indoc! {r#"
-    #[contract]
+const BALANCE_CONTRACT_INTERFACE: &str = indoc! {r#"
+    #[starknet::interface]
+    trait IBalance<T> {
+        // Returns the current balance.
+        fn get(self: @T) -> u128;
+        // Increases the balance by the given amount.
+        fn increase(ref self: T, a: u128);
+    }
+"#};
+
+const BALANCE_CONTRACT_IMPLEMENTATION: &str = indoc! {r#"
+    #[starknet::contract]
     mod Balance {
+        use traits::Into;
+
+        #[storage]
         struct Storage {
-            balance: felt252,
+            value: u128,
         }
 
-        // Increases the balance by the given amount.
-        #[external]
-        fn increase_balance(amount: felt252) {
-            balance::write(balance::read() + amount);
+        #[constructor]
+        fn constructor(ref self: ContractState, value_: u128) {
+            self.value.write(value_);
         }
 
-        // Returns the current balance.
-        #[view]
-        fn get_balance() -> felt252 {
-            balance::read()
+        #[external(v0)]
+        impl Balance of super::IBalance<ContractState> {
+            fn get(self: @ContractState) -> u128 {
+                self.value.read()
+            }
+            fn increase(ref self: ContractState, a: u128)  {
+                self.value.write( self.value.read() + a );
+            }
         }
     }
 "#};
 
+static BALANCE_CONTRACT: Lazy<String> = Lazy::new(|| {
+    format!(
+        "{}\n{}",
+        BALANCE_CONTRACT_INTERFACE, BALANCE_CONTRACT_IMPLEMENTATION
```

Comment:
Nit, these could be used directly in format I think.
```suggestion
        "{BALANCE_CONTRACT_INTERFACE}\n{BALANCE_CONTRACT_IMPLEMENTATION}",
```